services/sensorCachingService.js

```diff
@@ -0,0 +1,84 @@
+//Declaring sensors and creating the function(s) for it
+var sensors = {}
+
+/**
+ *Retrieve the last known temprature/humidity form a specific sensor
+ * @author Lars
+ * @param {entry} entry The sensor entry that you want to add to the cache
+ */
+function retrieveLastValue(entry) {
+    const key = getKey(entry);
+    if(sensors[key]){
+        return sensors[key].value
+    }
+    return null
+}
+
+/**
+ * Store the current value in the sensors obj
+ * @author Lars
+ * @param {entry} entry The sensor entry that you want to add to the cache
+ */
+function storeCurrentValue(entry){
+    sensors[getKey(entry)] = entry
+}
+
+/**
+ * Retrieve the maximum temprature/humidity for a specific floor.
+ * @author Lars
+ * @param {floor} floor The floor you want to recieve the max temp of.
+ */
+function getMaxTemp(floor){
+    var highestTemp
+    for(var element of sensorsFromFloor(floor)){
+        if(!highestTemp || element.value > highestTemp){
+            highestTemp = element.value
+        }
+    }
+    return highestTemp
+}
```

Comment:
This compares temperatures to humidity because `value `could be either temperature or humidity and no check for `parameter` is made.

Reply:
Btw, we do implement the policy that any bug needs to have a test proving that the bug was solved, right?

Reply:
Created a `BUGFIX - highest temprature function should not return a humidity value` test function that's now working.